codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
@Scottmitch currently the event in `Http2Connection.Listener` is `goingAway()`, which would get called in both cases. So the duplication of the inbound event already exists to some extent.  What would your preference be? To change `goingAway` to be `onGoAwaySent` and make the inbound GOAWAY exclusive to `Http2FrameListener.onGoAwayRead`?